src/cljs/main/broadfcui/page/workspace/notebooks/notebooks.cljs

```diff
@@ -264,25 +270,52 @@
                       (do ((:dismiss props)) ((:reload-after-delete props))) ;if success, update the table?
                       (swap! state assoc :server-error (get-parsed-response false))))})))})
 
+(react/defc- ClusterErrorViewer
+  ; Note cluster errors are only returned on "get cluster" requests to Leo, not "list cluster" requests.
+  ; Therefore we need issue an additional ajax call to Leo before we can display the errors.
+  {:component-did-mount
+   (fn [{:keys [this]}]
+     (this :-get-cluster-details))
+   :render
+   (fn [{:keys [state this props]}]
+     (let [{:keys [cluster-to-view-details server-error]} @state
+           {:keys [cluster-to-view]} props]
+       [modals/OKCancelForm
+        {:header "Cluster Error"
+         :dismiss (:dismiss props)
+         :ok-button {:text "Done"
+                     :onClick (:dismiss props)}
+         :show-cancel? false
+         :content
+         (react/create-element
+          [:div {:style {:width 700}}
+           (when-not cluster-to-view-details (blocker "Getting error details..."))
+           (when server-error [comps/ErrorViewer {:data-test-id "notebooks-error" :error server-error}])
```

Comment:
`ErrorViewer` already checks the existence of the error, so you can unwrap this `when`.